### PR TITLE
Prepare `v2.0.5` release

### PR DESCRIPTION
**Pull request by mossmaurice**
## Pre-Review Checklist for the PR Author

1. [x] ~~Add a second reviewer for complex new features or larger refactorings~~
1. [x] ~~Code follows the coding style of [CONTRIBUTING.md][contributing]~~
1. [x] ~~Tests follow the [best practice for testing][testing]~~
1. [x] Changelog updated [in the unreleased section][changelog] including API breaking changes
1. [x] Branch follows the naming format (`iox-123-this-is-a-branch`)
1. [x] Commits messages are according to this [guideline][commit-guidelines]
    - [x] Commit messages have the issue ID (`iox-#123 commit text`)
    - [x] Commit author matches [Eclipse Contributor Agreement][eca] (and ECA is signed)
1. [x] Update the PR title
   - Follow the same conventions as for commit messages
   - Link to the relevant issue
1. [x] Relevant issues are linked
1. [x] Add sensible notes for the reviewer
1. [x] All checks have passed (except `task-list-completed`)
1. [x] ~~All touched (C/C++) source code files from `iceoryx_hoofs` are added to `./clang-tidy-diff-scans.txt`~~
1. [x] Assign PR to reviewer

[commit-guidelines]: https://tbaggery.com/2008/04/19/a-note-about-git-commit-messages.html
[eca]: http://www.eclipse.org/legal/ECA.php
[contributing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/CONTRIBUTING.md#coding-style
[testing]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/concepts/best-practice-for-testing.md
[changelog]: https://github.com/eclipse-iceoryx/iceoryx/blob/master/doc/website/release-notes/iceoryx-unreleased.md

## Notes for Reviewer

* Prepare `v2.0.5` release

## Checklist for the PR Reviewer

- [x] Commits are properly organized and messages are according to the guideline
- [x] Code according to our coding style and naming conventions
- [x] Unit tests have been written for new behavior
- [x] Public API changes are documented via doxygen
- [x] Copyright owner are updated in the changed files
- [x] All touched (C/C++) source code files from `iceoryx_hoofs` have been added to `./clang-tidy-diff-scans.txt`
- [x] PR title describes the changes

## Post-review Checklist for the PR Author

1. [x] All open points are addressed and tracked via issues

## References

- Relates #2099

### CODE REVIEW

**Comment by mossmaurice** [comment]
ROS 2 CI:

* Linux [Build Status](http://ci.ros2.org/job/ci_linux/19907/)
* Linux-aarch64 [Build Status](http://ci.ros2.org/job/ci_linux-aarch64/14386/)
* Windows [Build Status](http://ci.ros2.org/job/ci_windows/20613/)

File: tools/ci/run-integration-test.sh [hunk 1]
```diff
@@ -29,18 +29,18 @@ cd "${WORKSPACE}"
...
-sudo sed -i '41 c\"'c'", "'cc'", "'cpp'", "'cxx'", "'h'", "'hh'", "'hpp'", "'hxx'", "'inl'", "'sh'"' /opt/ros/foxy/lib/python3.8/site-packages/ament_copyright/main.py
-ament_copyright --exclude LICENSE CONTRIBUTING.md tools/apache2_header.txt
+source /opt/ros/iron/setup.bash
+
+#msg "checking copyrights"
+#sudo rm -rf /opt/ros/iron/lib/python3.8/site-packages/ament_copyright/template/apache2_header.txt
+#sudo cp -rf tools/apache2_header.txt /opt/ros/iron/lib/python3.8/site-packages/ament_copyright/template/.
+## shellcheck disable=SC2026
+#sudo sed -i '41 c\"'c'", "'cc'", "'cpp'", "'cxx'", "'h'", "'hh'", "'hpp'", "'hxx'", "'inl'", "'sh'"' /opt/ros/iron/lib/python3.8/site-packages/ament_copyright/main.py
+#ament_copyright --exclude LICENSE CONTRIBUTING.md tools/apache2_header.txt
```
Comment (dkroenke) [inline tools/ci/run-integration-test.sh:43]:
```suggestion
msg "checking copyrights"
sudo rm -rf /opt/ros/iron/lib/python3.10/site-packages/ament_copyright/template/apache2_header.txt
sudo cp -rf tools/apache2_header.txt /opt/ros/iron/lib/python3.10/site-packages/ament_copyright/template/.
# shellcheck disable=SC2026
sudo sed -i '41 c\"'c'", "'cc'", "'cpp'", "'cxx'", "'h'", "'hh'", "'hpp'", "'hxx'", "'inl'", "'sh'"' /opt/ros/iron/lib/python3.10/site-packages/ament_copyright/main.py
ament_copyright --exclude LICENSE CONTRIBUTING.md tools/apache2_header.txt
```

Comment (dkroenke) [reply tools/ci/run-integration-test.sh:43]: Scratch that, it is better to be based on ROS 2 Humble here which is the LTS version.

**Review by dkroenke — APPROVED** [review]
LGTM.

**Pull request merged**